send-transaction-service/src/send_transaction_service.rs

```diff
@@ -391,7 +421,8 @@ impl SendTransactionService {
     ) -> ProcessTransactionsResult {
         let mut result = ProcessTransactionsResult::default();
 
-        let mut batched_transactions = HashSet::new();
```

Comment:
have we already deduplicated `transactions` by now?

Reply:
yes, because transactions come from `HashMap`